sbin/root-mount

```diff
@@ -149,18 +162,28 @@ if ! lvm vgchange -a y sailfish; then
 	exit 1
 fi
 
-# Check factory reset need.
+# Check factory reset need and read FS resize status.
 if mount -t ext4 $ROOTDEV $1; then
 	factory_reset_if_needed $1
+
+	if test -f "$1/$FS_RESIZED"; then
+		IS_FS_RESIZED=1
+	fi
+
 	umount $1
 fi
 
 # Check and resize root and home in case usage indicates first boot
-check_firstboot_resize
+if ! check_firstboot_resize; then
+	echo "First boot resize failed. Aborting boot." > /dev/kmsg
+	exit 1
```

Comment:
I've thought about it too... The question is what is better: trying to boot with some troubles on filesystem (and possibly corrupt it even more by writing to it) or refuse to boot, allowing a user to fix it from recovery mode?

User friendliness vs reliability and data integrity. :-)